mbrl/util/huggingface.py

```diff
@@ -0,0 +1,351 @@
+    :std_reward: standard deviation of the mean reward of the agent
+    """
+    metadata = {}
+    metadata["library_name"] = "stable-baselines3"
+    metadata["tags"] = [
+        env_id,
+        "deep-reinforcement-learning",
+        "reinforcement-learning",
+        "stable-baselines3",
+    ]
+
+    # Add metrics
+    eval = metadata_eval_result(
+        model_pretty_name=model_name,
+        task_pretty_name="reinforcement-learning",
+        task_id="reinforcement-learning",
+        metrics_pretty_name="mean_reward",
+        metrics_id="mean_reward",
+        metrics_value=f"{mean_reward:.2f} +/- {std_reward:.2f}",
+        dataset_pretty_name=env_id,
+        dataset_id=env_id,
+    )
+
+    # Merges both dictionaries
+    metadata = {**metadata, **eval}
+
+    return metadata
+
+
+def _generate_model_card(
+    model_name: str, env_id: str, mean_reward: float, std_reward: float
+) -> Tuple[str, Dict[str, Any]]:
+    """
+    Generate the model card for the Hub
+    :param model_name: name of the model
+    :env_id: name of the environment
+    :mean_reward: mean reward of the agent
+    :std_reward: standard deviation of the mean reward of the agent
+    """
+    # Step 1: Select the tags
+    metadata = generate_metadata(model_name, env_id, mean_reward, std_reward)
+
+    # Step 2: Generate the model card
+    model_card = f"""
+# **{model_name}** Agent playing **{env_id}**
+This is a trained model of a **{model_name}** agent playing **{env_id}**
+using the [stable-baselines3 library](https://github.com/DLR-RM/stable-baselines3).
+"""
+
+    model_card += """
+## Usage (with Stable-baselines3)
+TODO: Add your code
+```python
+from stable_baselines3 import ...
+from huggingface_sb3 import load_from_hub
+...
+```
+"""
+
+    return model_card, metadata
+
+
+def _save_model_card(
+    local_path: Path, generated_model_card: str, metadata: Dict[str, Any]
+):
+    """Saves a model card for the repository.
+    :param local_path: repository directory
+    :param generated_model_card: model card generated by _generate_model_card()
+    :param metadata: metadata
+    """
+    readme_path = local_path / "README.md"
+    readme = ""
+    if readme_path.exists():
+        with readme_path.open("r", encoding="utf8") as f:
+            readme = f.read()
+    else:
+        readme = generated_model_card
+
+    with readme_path.open("w", encoding="utf-8") as f:
+        f.write(readme)
+
+    # Save our metrics to Readme metadata
+    metadata_save(readme_path, metadata)
+
+
+def _add_logdir(local_path: Path, logdir: Path):
+    """Adds a logdir to the repository.
+    :param local_path: repository directory
+    :param logdir: logdir directory
+    """
+    if logdir.exists() and logdir.is_dir():
+        # Add the logdir to the repository under new dir called logs
+        repo_logdir = local_path / "logs"
+
+        # Delete current logs if they exist
+        if repo_logdir.exists():
+            shutil.rmtree(repo_logdir)
```

Comment:
Here `repo_logdir` is a tmp dir (at least when called from `package_to_hub`.
So deleting it locally will not delete it on the Hub. A possibility is to use `delete_folder` to delete the logs folder on the Hub before uploading the new one.